### PR TITLE
Mzieg 2048px

### PR DESCRIPTION
**Pull request by mzieg**
Hi Trent, I'll still be putzing about with test apps and driver tweaks for our 2048-pixel units for awhile, but this particular bugfix seemed worth pushing out now.  Apparently it really screws up comms if you try to read detector temperature from an ambient spectrometer.

### CODE REVIEW

File: WasatchNET/Spectrometer.cs [hunk 1]
```diff
@@ -824,6 +852,8 @@ public ushort detectorTemperatureRaw
             {
                 if (isSiG)
                     return 0;
+                if (!eeprom.hasCooling)
```
Comment (mzieg) [inline WasatchNET/Spectrometer.cs:855]: This was a surprisingly important bug.

File: Setup32/Setup32.vdproj [hunk 2]
```diff
@@ -535,7 +535,7 @@
             {
             "AssemblyRegister" = "3:1"
             "AssemblyIsInGAC" = "11:FALSE"
-            "AssemblyAsmDisplayName" = "8:WasatchNET, Version=2.2.8.0, Culture=neutral, processorArchitecture=x86"
+            "AssemblyAsmDisplayName" = "8:WasatchNET, Version=2.2.9.0, Culture=neutral, processorArchitecture=x86"
```
Comment (trestoh) [inline Setup32/Setup32.vdproj:538]: Are we on 9 or 10?

File: WasatchNET/Spectrometer.cs [hunk 3]
```diff
@@ -2209,7 +2250,7 @@ public virtual double[] getSpectrum(bool forceNew=false)
                         sum[px] -= dark_[px];
 
                 // this should be enough to update the cached value
-                if (readTemperatureAfterSpectrum)
+                if (readTemperatureAfterSpectrum && eeprom.hasCooling)
```
Comment (trestoh) [inline WasatchNET/Spectrometer.cs:2253]: should be redundant now that we have lower level guard

File: WasatchNET/Spectrometer.cs [hunk 4]
```diff
@@ -2244,8 +2285,15 @@ public bool sendSWTrigger()
         /// If the caller doesn't want to block on this, they can always change
         /// integrationTimeMS within a Task.Run closure.
         /// </remarks>
+        ///
+        /// <todo>
+        /// Consider simply disabling this feature (returning from this function)
+        /// if autoTrigger is disabled or using HW triggering.  In such cases, if
+        /// the user wants a throwaway, they can generate it themselves.
+        /// </todo>
```
Comment (trestoh) [inline WasatchNET/Spectrometer.cs:2293]: WPSC was already performing this in software. I'm admittedly not very familiar with our various triggering setups

File: Setup32/Setup32.vdproj (hunk 2 above)
Comment (mzieg) [reply Setup32/Setup32.vdproj:538]: I saw that too...I'm honestly not sure how that happened.  The "AssemblyInfo.cs" file had the correct value.  I'll check in the merge.

File: WasatchNET/Spectrometer.cs (hunk 3 above)
Comment (mzieg) [reply WasatchNET/Spectrometer.cs:2253]: I was upset enough that redundant felt good...

File: Setup32/Setup32.vdproj (hunk 2 above)
Comment (mzieg) [reply Setup32/Setup32.vdproj:538]: Gah.  That 2.2.8 -> 2.2.9 referred to the version of LibUsbDotNet.dll, NOT Wasatch.NET, even though we were changing between almost the exact same version numbers :-(

**Pull request merged**